modules/auxiliary/spoof/nbns/nbns_response.rb

```diff
@@ -44,108 +47,142 @@ def initialize
     ])
 
     register_advanced_options([
-      OptBool.new('Debug', [ false, "Determines whether incoming packet parsing is displayed", false])
+      OptBool.new('DEBUG', [ false, "Determines whether incoming packet parsing is displayed", false])
     ])
 
     deregister_options('RHOST', 'PCAPFILE', 'SNAPLEN', 'FILTER')
+    self.thread = nil
+    self.sock = nil
   end
 
-  def run
-    check_pcaprub_loaded() # Check first since otherwise this is all for naught
-    # MacOS X workaround
-    ::Socket.do_not_reverse_lookup = true
+  def dispatch_request(packet, rhost, src_port)
+    rhost = ::IPAddr.new(rhost)
+     # `recvfrom` (on Linux at least) will give us an ipv6/ipv4 mapped
+    # addr like "::ffff:192.168.0.1" when the interface we're listening
+    # on has an IPv6 address. Convert it to just the v4 addr
+    if rhost.ipv4_mapped?
+      rhost = rhost.native
+    end
 
-    @sock = ::UDPSocket.new()
-    @sock.setsockopt(::Socket::SOL_SOCKET, ::Socket::SO_REUSEADDR, 1)
-    @sock.bind('', 137) # couldn't specify srv host because it missed broadcasts
+    # Convert to string
+    rhost = rhost.to_s
 
-    @run = true
+    spoof = ::IPAddr.new(datastore['SPOOFIP'])
```

Comment:
This should be used below instead of calculating `hton` manually:  see https://github.com/rapid7/metasploit-framework/pull/4827/files#r26622744